File: aluguel-db/src/main/java/com/nortetec/imovel/Imovel.java
package com.nortetec.imovel;


import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serial;
import java.io.Serializable;

@Document(collection = "tab_imovel")
public class Imovel implements Serializable {
    @Serial
    private static final long serialVersionUID = -5666426053006243716L;
    private String endereco;
    private Double valorAluguel;

    public Imovel(String endereco, Double valorAluguel) {
        this.endereco = endereco;
        this.valorAluguel = valorAluguel;
    }

    public Imovel() {
    }

    public String getEndereco() {
        return endereco;
    }

    public Double getValorAluguel() {
        return valorAluguel;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Imovel imovel = (Imovel) o;

        if (!endereco.equals(imovel.endereco)) return false;
        return valorAluguel.equals(imovel.valorAluguel);
    }

    @Override
    public int hashCode() {
        int result = endereco.hashCode();
        result = 31 * result + valorAluguel.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "Imovel{" +
                "endereco='" + endereco + '\'' +
                ", valorAluguel=" + valorAluguel +
                '}';
    }
}

File: src/main/kotlin/com/nortetec/imovel/Imovel.kt
package com.nortetec.imovel

import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "tab_imovel")
data class Imovel(
    val endereco: String,
    val valorAluguel: Double,
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Imovel

        if (endereco != other.endereco) return false
        return valorAluguel == other.valorAluguel
    }

    override fun hashCode(): Int {
        var result = endereco.hashCode()
        result = 31 * result + valorAluguel.hashCode()
        return result
    }

    override fun toString(): String {
        return "Imovel(endereco='$endereco', valorAluguel=$valorAluguel)"
    }
}